### DISCUSSION TITLE
Using an ifm3d O3D313 camera with the OVP800

### DISCUSSION THREAD

**Discussion by niallomahony93** [ifm/ifm3d#415] [General]
I wish to use the OVP800 to do some processing to detect some targets viewed by an O3D313 camera (instead of the o3r camera heads). Is this possible? I suppose one would just need to change the ifm3d version to 0.20... to support that camera instead of the 03r heads. 

P.s. It may be possible to do the detection entirely on the o3d313 if my algorithm does not have too much complexity but I have not come across any documentation on how to put custom firmware on this camera.

**Comment by lola-masson** [comment]
Hi @niallomahony93,

You should be able to connect your O3D through the Eth1 port. Then, package your application in a docker container, and run this docker container on the OVP800. Here's [some doc on the Ethernet interface](https://ifm3d.com/documentation/Technology/VPU/ethernet.html), and [some docs to get started with Docker](https://ifm3d.com/documentation/SoftwareInterfaces/Docker/docker.html). In your docker container, you can install whatever ifm3d version you need.

As for deploying applications on the O3D, I am not sure there is much space left by the firmware but maybe @graugans can give us more details?

I am curious to hear: why are you trying to use the O3D instead of an O3R camera head? Is this because you have existing code that uses the O3D? Or is this a new project? Feel free to email us at support.robotics@ifm.com if you want to share the details of your application in a non-public way.

**Reply by graugans** [reply]
Hej @niallomahony93, 

please take care that the default IP address of both devices is `192.168.0.69`, if this is solved and the IP range of the second Ethernet interface does not overlap the one from the first interface and the O3D3xx is within the Network of the second Ethernet there shall be no issue as @lola-masson pointed out.

In regards of bringing software to the O3D3xx please get in touch with the support mail address @lola-masson already pointed out. One thing to keep in mind, there is no Docker support on O3D3xx!
